src/senselab/audio/tasks/features_extraction/torchaudio.py

```diff
@@ -168,6 +168,33 @@ def extract_mel_filter_bank_from_audios(
     return mel_filter_banks
 
 
+def extract_mel_filter_bank_from_spectrograms(
+    spectrograms: List[Dict[str, torch.Tensor]],
+    sampling_rate: int,
+    n_mels: int = 128,
+) -> List[Dict[str, torch.Tensor]]:
+    """Extract mel filter bank from a list of audio objects.
+
+    Args:
+        spectrograms (List[torch.Tensor]): List of spectrograms.
+        sampling_rate (int): Sampling rate of the audio.
+        n_mels (int): Number of mel filter banks. Default is 128.
+
+    Returns:
+        List[Dict[str, torch.Tensor]]: List of Dict objects containing mel filter banks.
+    """
+    mel_filter_banks = []
+    for spectrogram in spectrograms:
+        try:
+            melscale_transform = torchaudio.transforms.MelScale(
+                sample_rate=sampling_rate, n_mels=n_mels, n_stft=spectrogram["spectrogram"].shape[0]
```

Comment:
can you check whether this is (2*shape[0] + 1) or this?

Reply:
It should be fine the way it is. [torch.transforms.Spectrogram](https://pytorch.org/audio/main/generated/torchaudio.transforms.Spectrogram.html) returns a torch.tensor of size (freq, time), where freq is `n_fft // 2 + 1`, where `n_fft` is the number of Fourier bins, and time is the number of window hops (`n_frame`). 
[torchaudio.transforms.MelScale](https://pytorch.org/audio/main/generated/torchaudio.transforms.MelScale.html) wants `n_stft` , which is exactly `n_fft // 2 + 1`. hence, `spectrogram["spectrogram"].shape[0]` works as expected in our case.

Reply:
thanks for looking into this.